### DISCUSSION TITLE
ImportError: cannot import name 'VOCABS' from partially initialized module 'doctr.datasets'

### DISCUSSION THREAD

**Discussion by yerin723** [mindee/doctr#1294] [General]
I'm trying to make a model out of the training data I labeled.
However, the following error keeps occurring.
ImportError: cannot import name 'VOCABS' from partially initialized module 'doctr.datasets' (most likely due to a circular import) (C:\Users\PIAI\Desktop\ocrproject\doctr\doctr\datasets\__init__.py)
Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...

python "C:\Users\PIAI\Desktop\ocrproject\doctr\references\detection\train_pytorch.py" "C:\Users\PIAI\Desktop\test" "C:\Users\PIAI\Desktop\val" db_resnet50 -- epochs 5 --device 0

An error occurs when executing the above command.
Please tell me the solution!

**Comment by dvivarelli** [comment]
Same problem here

**Reply by felixdittrich92** [reply]
Hi @dvivarelli :wave:,

Are you also on a windows machine ?

**Reply by dvivarelli** [reply]
yes, windows 11

**Reply by felixdittrich92** [reply]
Have you already tried to create a new venv / conda env and reinstalled ?
Unfortunately i can't test it right now on a windows machine, but on Linux there are no issues.

**Reply by hanshupe007** [reply]
I have the same problem here, using Windows and Python 3.9.

According to the traceback below, the __init__.py in doctr.datasets (doctr.datasets import VOCABS) itself loads indirectly VOCABS from mobilenet\pytorch.py, which leads to the circular import error. 
```

Traceback (most recent call last):
  File "C:\xx\doctr\references\recognition\train_pytorch.py", line 33, in <module>
    from doctr.datasets import VOCABS
  File "C:\xx\doctr\doctr\datasets\__init__.py", line 3, in <module>
    from .generator import *
  File "C:\xx\doctr\doctr\datasets\generator\__init__.py", line 6, in <module>
    from .pytorch import *  # type: ignore[assignment]
  File "C:\xx\doctr\doctr\datasets\generator\pytorch.py", line 8, in <module>
    from .base import _CharacterGenerator, _WordGenerator
  File "C:\xx\doctr\doctr\datasets\generator\base.py", line 14, in <module>
    from ..datasets import AbstractDataset
  File "C:\xx\doctr\doctr\datasets\datasets\__init__.py", line 6, in <module>
    from .pytorch import *  # type: ignore[assignment]
  File "C:\xx\doctr\doctr\datasets\datasets\pytorch.py", line 15, in <module>
    from .base import _AbstractDataset, _VisionDataset
  File "C:\xx\doctr\doctr\datasets\datasets\base.py", line 16, in <module>
    from ...models.utils import _copy_tensor
  File "C:\xx\doctr\doctr\models\__init__.py", line 1, in <module>
    from .classification import *
  File "C:\xx\doctr\doctr\models\classification\__init__.py", line 1, in <module>
    from .mobilenet import *
  File "C:\xx\doctr\doctr\models\classification\mobilenet\__init__.py", line 6, in <module>
    from .pytorch import *
  File "C:\xx\doctr\doctr\models\classification\mobilenet\pytorch.py", line 13, in <module>
    from doctr.datasets import VOCABS
ImportError: cannot import name 'VOCABS' from partially initialized module 'doctr.datasets' (most likely due to a circular import) (C:\xx\doctr\doctr\datasets\__init__.py)
```

**Reply by felixdittrich92** [reply]
Hi @dvivarelli @hanshupe007 :wave:,

I have had the chance to test it on a windows machine.
Unfortunately i was not able to reproduce it...

Workflow:
- Setup miniconda -> create env
- Clone the repo
- Install pytorch via conda (optional) - on windows it's easier with conda because of precompiled cuda/cudnn - GPU support
- open repo in VSCode -> pip3 install -e ."[torch]" -> pip3 install wandb
- move into recognition folder
- `python train_pytorch.py crnn_mobilenet_v3_small`

**Reply by hanshupe007** [reply]
I didn't use conda, but WinPython, did you also use Python 3.9? Looking at the traceback the error makes sense to me, as the __init__py is triggered by the datasets import, and itself imports VOCABS, not sure why it doesn't throw the circular import error on other systems.

**Reply by hanshupe007** [reply]
While I couldn't resolve the error on my Windows/Python 3.9 environment, it worked on Linux/3.10.

**Reply by felixdittrich92** [reply]
Hi @hanshupe007 :wave:,

Thanks for the feedback.
Good to hear that it works now.
But still strange - because as mentioned i haven't had any trouble also on Win :/
maybe a issue with WinPython .. 
In general i can only suggest conda (miniconda) - makes it much easier to handle venvs
